crates/hyperswitch_connectors/src/connectors/nomupay/transformers.rs

```diff
@@ -0,0 +1,771 @@
+    Processed,
+    Failed,
+    #[default]
+    Processing,
+    Scheduled,
+    PendingAccountActivation,
+    PendingTransferMethodCreation,
+    PendingAccountKyc,
+}
+
+impl From<NomupayPaymentStatus> for common_enums::AttemptStatus {
+    fn from(item: NomupayPaymentStatus) -> Self {
+        match item {
+            NomupayPaymentStatus::Pending => Self::Pending,
+            NomupayPaymentStatus::Processed => Self::Charged,
+            NomupayPaymentStatus::Failed => Self::Failure,
+            NomupayPaymentStatus::Processing => Self::Authorizing,
+            NomupayPaymentStatus::Scheduled => Self::Pending,
+            NomupayPaymentStatus::PendingAccountActivation => Self::AuthenticationPending,
+            NomupayPaymentStatus::PendingTransferMethodCreation => Self::PaymentMethodAwaited,
+            NomupayPaymentStatus::PendingAccountKyc => Self::AuthenticationPending,
+        }
+    }
+}
+
+impl From<NomupayPaymentStatus> for PayoutStatus {
+    fn from(item: NomupayPaymentStatus) -> Self {
+        match item {
+            NomupayPaymentStatus::Pending => Self::Pending,
+            NomupayPaymentStatus::Processed => Self::Success,
+            NomupayPaymentStatus::Failed => Self::Failed,
+            NomupayPaymentStatus::Processing => Self::Pending,
+            NomupayPaymentStatus::Scheduled => Self::Pending,
+            NomupayPaymentStatus::PendingAccountActivation => Self::Pending,
+            NomupayPaymentStatus::PendingTransferMethodCreation => Self::Pending,
+            NomupayPaymentStatus::PendingAccountKyc => Self::Pending,
+        }
+    }
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct NomupayPaymentsResponse {
+    status: NomupayPaymentStatus,
+    id: String,
+}
+
+impl<F, T> TryFrom<ResponseRouterData<F, NomupayPaymentsResponse, T, PaymentsResponseData>>
+    for RouterData<F, T, PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<F, NomupayPaymentsResponse, T, PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: common_enums::AttemptStatus::from(item.response.status),
+            response: Ok(PaymentsResponseData::TransactionResponse {
+                resource_id: ResponseId::ConnectorTransactionId(item.response.id),
+                redirection_data: Box::new(None),
+                mandate_reference: Box::new(None),
+                connector_metadata: None,
+                network_txn_id: None,
+                connector_response_reference_id: None,
+                incremental_authorization_allowed: None,
+                charge_id: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// REFUND :
+// Type definition for RefundRequest
+#[derive(Default, Debug, Serialize)]
+pub struct NomupayRefundRequest {
```

Comment:
Where is this used? Can we remove redundant code?

Reply:
removed done